### PR TITLE
Gitlab Tests Restructuring

### PR DESCRIPTION
**Pull request by sidpremkumar**

### CODE REVIEW

**Comment by amarza-rh** [comment]
An error in python3:

```
test/gitlab_tests/test_gitlab.py:305:
...
        except SSLError as e:
>           log.exception('Error during authentification: %s', e.message)
E           AttributeError: 'SSLError' object has no attribute 'message'
```

The `message` attribute has been deprecated in [PEP-352](https://www.python.org/dev/peps/pep-0352)

Please run tests both from python2 and python3 if the purpose is to maintain compatibility across pythons.

File: reviewrot/gitlabstack.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def request_reviews(self, user_name, repo_name=None, age=None,
         try:
             gl.auth()
         except SSLError as e:
-            log.exception('Error during authentification: %s', e.message)
```
Comment (sidpremkumar) [inline reviewrot/gitlabstack.py]: This used to cause an error in python3 ('AttributeError: 'SSLError' object has no attribute 'message''). Changing it to __str__() fixes this.

Comment (amarza-rh) [reply reviewrot/gitlabstack.py]: Does `str(e)` also work here? (instead of `e.__str__()`)

**Comment by amarza-rh** [comment]
You might also want to cover the following with a test from `gitlabstack.py`

```python
            except ValueError: 
                mr_date = datetime.datetime.strptime( 
                    mr.created_at, '%Y-%m-%dT%H:%M:%SZ') 
                mr_updated_date = datetime.datetime.strptime( 
                    mr.updated_at, '%Y-%m-%dT%H:%M:%SZ') 
```

File: reviewrot/gitlabstack.py (hunk 1 above)
Comment (sidpremkumar) [reply reviewrot/gitlabstack.py]: Yes, I've just changed it to str(e)

**Comment by amarza-rh** [comment]
LGTM, thanks for your contribution :)

**Pull request merged**